lib/gather/gather-telemetry.js

```diff
@@ -28,235 +29,24 @@ module.exports = async function gatherTelemetry(url, puppeteerArgs = {}) {
       if (SKIPPED_MODULES.includes(modulePath)) {
         continue;
       }
+      if (window.gatherFn && typeof window.gatherFn === 'function') {
+        try {
+          let module = require(modulePath);
 
-      try {
-        let module = require(modulePath);
-
-        if (module && module.default && module.default.proto) {
-          let defaultProto = module.default.proto();
-
-          telemetry[modulePath] = parseMeta(Ember.meta(defaultProto));
-        }
-      } catch (error) {
-        // log the error, but continue
-        window.logErrorInNodeProcess(`error evaluating \`${modulePath}\`: ${error.message}`);
-      }
-    }
-
-    /**
-     * Compares the object with types of Ember objects
-     *
-     * @param {Object} object
-     * @returns {String} type
-     */
-    function getType(object) {
-      const types = [
-        'Application',
-        'Controller',
-        'Route',
-        'Component',
-        'Service',
-        'Helper',
-        'Router',
-        'Engine',
-      ];
-      return types.find(type => Ember[type] && object instanceof Ember[type]) || 'EmberObject';
-    }
-
-    /**
-     * Parses ember meta data object and collects the runtime information
-     *
-     * @param {Object} meta
-     * @returns {Object} data - Parsed metadata for the ember object
-     * @returns {String[]} data.computedProperties - list of computed properties
-     * @returns {String[]} data.getters - list of ES5 getters
-     * @returns {String[]} data.observedProperties - list of observed properties
-     * @returns {Object} data.observerProperties - list of observer properties
-     * @returns {Object} data.offProperties - list of observer properties
-     * @returns {String[]} data.overriddenActions - list of overridden actions
-     * @returns {String[]} data.overriddenProperties - list of overridden properties
-     * @returns {String[]} data.ownProperties - list of object's own properties
-     * @returns {String} data.type - type of ember object
-     * @returns {Object} data.unobservedProperties - list of unobserved properties
-     */
-    function parseMeta(meta = {}) {
-      if (!meta || !meta.source) {
-        return {};
-      }
-      const { source } = meta;
-      const type = getType(source);
-
-      const ownProperties = Object.keys(source).filter(key => !['_super', 'actions'].includes(key));
+          if (module && module.default && module.default.proto) {
```

Comment:
I think we should probably take this guard out too, and pass the `module` variable in to the gather function (it should be possible for a codemod to look at non-default exports, which isn't possible with this current logic).

Reply:
I’m temporarily using this as a testing function until I’m confident in the general implementation. I was going to get rid If it.

Reply:
Oops commented on the wrong one!